src/transformers/modeling_utils.py

```diff
@@ -2449,6 +2451,15 @@ def _find_mismatched_keys(
                 gc.collect()
 
             if offload_index is not None and len(offload_index) > 0:
+                if model != model_to_load:
+                    # We need to add the prefix of the base model
```

Comment:
(nit) Maybe make the comment a bit more explicit? Took me a bit to figure out what is going on here

```suggestion
                    # We need to add the prefix of the base model `model_to_load` to be able to correctly the weights into the task model `model`
```